x-pack/plugins/enterprise_search/server/lib/indices/fetch_indices.ts

```diff
@@ -76,24 +83,47 @@ export const fetchIndices = async (
     index: indexPattern,
     metric: ['docs', 'store'],
   });
-  const resultIndices = indicesNames
+
+  // TODO: make multiple batched requests if indicesNames.length > SOMETHING
+  const { index: indexPrivileges } = await client.asCurrentUser.security.hasPrivileges({
+    index: [
+      {
+        names: indexAndAliasNames,
+        privileges: ['read', 'manage'],
+      },
+    ],
+  });
+
+  return indicesNames
     .map((indexName: string) => {
       const indexData = totalIndices[indexName];
       const indexStats = indicesStats[indexName];
       return mapIndexStats(indexData, indexStats, indexName);
     })
-    .flatMap(({ name, aliases, ...engineData }) => {
+    .flatMap(({ name, aliases, ...indexData }) => {
       // expand aliases and add to results
-      const engines = [];
-      engines.push({ name, ...engineData });
+      const indicesAndAliases = [] as ElasticsearchIndex[];
+      indicesAndAliases.push({
+        name,
+        alias: false,
+        privileges: indexPrivileges[name],
+        ...indexData,
+      });
 
       aliases.forEach((alias) => {
-        engines.push({ name: alias, ...engineData });
+        indicesAndAliases.push({
+          name: alias,
+          alias: true,
+          privileges: indexPrivileges[alias],
+          ...indexData,
+        });
       });
-      return engines;
-    });
-
-  // The previous step could have added indices that don't match the index pattern, so filter those out again
-  // We wildcard RegExp the pattern unless user provides a more specific regex
-  return indexRegExp ? resultIndices.filter(({ name }) => name.match(indexRegExp)) : resultIndices;
+      return indicesAndAliases;
+    })
+    .filter(
+      ({ name }, index, array) =>
+        // make list of aliases unique since we add an alias per index above
```

Comment:
Sidenote: love this comment 'cause I was staring at this earlier trying to figure out why we were doing this :)